lgc/CMakeLists.txt

```diff
@@ -62,6 +64,20 @@ option(LLPC_ENABLE_WERROR "Build LLPC with more errors" OFF)
 include(../cmake/CompilerFlags.cmake)
 set_compiler_options(LLVMlgc ${LLPC_ENABLE_WERROR})
 
+### TableGen for LGC dialect ###########################################################################################
+
+include_directories(../imported/llvm-dialects/include)
+
+set(LGC_TABLEGEN_EXE llvm-dialects-tblgen)
+set(LGC_TABLEGEN_TARGET llvm-dialects-tblgen)
```

Comment:
There seems to be an `add_tablegen(llvm-dialects-tblgen llvm-dialects-tblgen)` function that handles some cross-compilation stuff, is there a reason we don’t use it?

Reply:
It is meant for building tablegen tools, as in, it will actually add the executable target for building the tablegen tool. As a side-effect it sets some useful variables, but those didn't end up in the correct scope when I tried it.